### ISSUE TITLE
[FORM] DateTimeToLocalizedStringTransformer setTimeZone is missing

### ISSUE THREAD

**Issue by mfuehrer82** [symfony/symfony#19981]
Hey,

in the DateTimeToLocalizedStringTransformer version 2.8.11 the timezone is not set in the reverseTransform method. 

The value of $timestamp at line 124 is in UTC. 
```
$timestamp = $this->getIntlDateFormatter($dateOnly)->parse($value);
```
My outputTimezone and inputTimeZone are 'Europe/Berlin' so they are equal. 
But the timezone will only be set 
```        
if ($this->outputTimezone !== $this->inputTimezone) {
       $dateTime->setTimezone(new \DateTimeZone($this->inputTimezone));
}  
```
In version 3.* it seems to be fixed. 

Thank you for watching this.

**Comment by jordisala1991** [comment]
Had the same problem, it is fixed on 2.8-dev.

This bug appeared on 2.8.10, so use 2.8.9 until they release 2.8.12

**Comment by Jean85** [comment]
This is the same of #19721, and it's already fixed. I'm waiting for 2.8.12 too.
